elide-core/src/main/java/com/yahoo/elide/core/exceptions/ForbiddenAccessException.java

```diff
@@ -20,20 +20,22 @@
     private static final long serialVersionUID = 1L;
 
     @Getter private final Optional<Expression> expression;
+    @Getter private final Optional<Expression.EvaluationMode> evaluationMode;
 
     public ForbiddenAccessException(String message) {
         super(HttpStatus.SC_FORBIDDEN, null, message);
         this.expression = Optional.empty();
+        this.evaluationMode = Optional.empty();
     }
 
-    public ForbiddenAccessException(String message, Expression expression) {
-        super(HttpStatus.SC_FORBIDDEN, null, message);
+    public ForbiddenAccessException(String message, Expression expression, Expression.EvaluationMode mode) {
+        super(HttpStatus.SC_FORBIDDEN, null, message + ": " + expression);
```

Comment:
This does a string manipulation for every ForbiddenAccess which include each filtered element in a large collection.